android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/HereYouGoService.java

```diff
@@ -0,0 +1,332 @@
+/*
+HereYouGoService.java
+
+Shares URLs via bluetooth
+Also interfaces with PWS to shorten URLs that are too long for Eddystone URLs
+Lastly, it surfaces a persistent notification whenever a URL is currently being broadcast
+*/
+
+package org.physical_web.physicalweb;
+import android.content.BroadcastReceiver;
+import android.content.IntentFilter;
+
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.bluetooth.BluetoothManager;
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.BluetoothManager;
+import android.bluetooth.le.AdvertiseCallback;
+import android.bluetooth.le.AdvertiseData;
+import android.bluetooth.le.AdvertiseSettings;
+import android.bluetooth.le.BluetoothLeAdvertiser;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Binder;
+import android.os.Handler;
+import android.os.IBinder;
+import android.os.ParcelUuid;
+import android.preference.PreferenceManager;
+import android.support.v4.app.NotificationManagerCompat;
+import android.support.v4.app.TaskStackBuilder;
+import android.support.v4.app.NotificationCompat;
+import android.util.Log;
+import android.widget.Toast;
+import java.util.Random;
+
+public class HereYouGoService extends Service
+        implements PwsClient.ShortenUrlCallbacks,
+        PwsClient.ResolveScanCallback {
+
+    private static final String TAG = "UrlSharerService";
+    private BluetoothManager mBluetoothManager;
+    private BluetoothLeAdvertiser mBluetoothLeAdvertiser;
+    private static final ParcelUuid URI_BEACON_UUID = ParcelUuid.fromString("0000FEAA-0000-1000-8000-00805F9B34FB");
```

Comment:
This seems like it belongs in the advertising utils...but don't move it just yet.  Let's talk about it after you make the other changes.